Deploy-CAM.ps1

```diff
@@ -129,8 +137,8 @@ function Register-CAM()
 	)
 
     $camDeploymentInfo = @{} #start with an empty hash map
-	$camRegistrationError = ""
```

Comment:
But it's used outside of the for loop below... Why the change?

Reply:
if it is outside of the loop, for example, at first iteration, there is error and it get the value, at the second iteration, it processed successful. Exit the loop, because the value is not clear. so the if statement if($camRegistrationError) {throw $camRegistrationError} executed and it throw exception. This will break your initial purpose of the retrying loop

Reply:
Good catch! I think I'd leave both in though. I guess it doesn't really matter and maybe 'my C++' is showing but would be nice to have the variable declared/initialized outside of the loop if it's consumed outside of the loop. A style thing that wouldn't affect functionality in this case... Thoughts?

Reply:
I would declare the variable outside the loop and reset the value in the loop. so make the code more clear. No sure the powershell variable scope.

Reply:
Sounds good. variable scope is 'within function' for local variables so variables within loops persist past the loops, unlike C++ or many other languages.